addons/point_of_sale/static/src/app/screens/scale_screen/scale_screen.js

```diff
@@ -45,79 +43,45 @@ export class ScaleScreen extends Component {
             return;
         }
         this.state.weight = await this.hardwareProxy.readScale();
+        this.pos.setScaleWeight(this.state.weight);
         setTimeout(() => this._setWeight(), 500);
     }
     get netWeight() {
-        const weight = round_pr(this.state.weight || 0, this.productUnit.rounding);
+        const weight = round_pr(this.state.weight || 0, this.props.uomRounding);
         const weightRound = weight.toFixed(
-            Math.ceil(Math.log(1.0 / this.productUnit.rounding) / Math.log(10))
+            Math.ceil(Math.log(1.0 / this.props.uomRounding) / Math.log(10))
         );
         return weightRound - parseFloat(this.state.tare);
     }
-    get _activePricelist() {
-        const current_order = this.pos.get_order();
-        let current_pricelist = this.pos.config.pricelist_id;
-        if (current_order) {
-            current_pricelist = current_order.pricelist_id;
-        }
-        return current_pricelist;
-    }
+
     get productWeightString() {
         const defaultstr = (this.state.weight || 0).toFixed(3) + " Kg";
-        if (!this.props.product) {
+        const uom = this.props.uomName;
+        if (!uom) {
             return defaultstr;
         }
-        const unit = this.productUnit;
-        if (!unit) {
-            return defaultstr;
-        }
-        const weight = round_pr(this.state.weight || 0, unit.rounding);
-        let weightstr = weight.toFixed(Math.ceil(Math.log(1.0 / unit.rounding) / Math.log(10)));
-        weightstr += " " + unit.name;
+        const weight = round_pr(this.state.weight || 0, this.props.uomRounding);
+        let weightstr = weight.toFixed(
+            Math.ceil(Math.log(1.0 / this.props.uomRounding) / Math.log(10))
```

Comment:
isn't there some ready made function to use instead of this ? For example in `web/core/utils/numbers.js` ?